tests/unit/python/foglamp/tasks/purge/test_purge.py

```diff
@@ -0,0 +1,256 @@
+            return ""
+
+        mockStorageClient = MagicMock(spec=StorageClient)
+        mockAuditLogger = AuditLogger(mockStorageClient)
+
+        with patch.object(FoglampProcess, '__init__'):
+            with patch.object(mockAuditLogger, "__init__", return_value=None):
+                p = Purge(loop=event_loop)
+                p._logger = logger
+                p._logger.info = MagicMock()
+                p._logger.error = MagicMock()
+                p._storage = MagicMock(spec=StorageClient)
+                p._readings_storage = MagicMock(spec=ReadingsStorageClient)
+                audit = p._audit
+                with patch.object(p._readings_storage, 'purge', side_effect=self.store_purge):
+                    with patch.object(audit, 'information', return_value=mock_audit_info()):
+                        assert expected_return == p.purge_data(conf)
+                        p._logger.error.assert_called_with('Purge failed: %s', '409 Conflict')
+
+    @pytest.mark.parametrize("conf, expected_error_key",
+                             [({"retainUnsent": {"value": "True"}, "age": {"value": "bla"}, "size": {"value": "0"}},
+                               "age"),
+                              ({"retainUnsent": {"value": "True"}, "age": {"value": "0"}, "size": {"value": "bla"}},
+                               "size")])
+    def test_purge_data_invalid_conf(self, event_loop, conf, expected_error_key):
+        """Test that purge_data raises exception when called with invalid configuration"""
+
+        @asyncio.coroutine
+        def mock_audit_info():
+            return ""
+
+        mockStorageClient = MagicMock(spec=StorageClient)
+        mockAuditLogger = AuditLogger(mockStorageClient)
+
+        with patch.object(FoglampProcess, '__init__'):
+            with patch.object(mockAuditLogger, "__init__", return_value=None):
+                p = Purge(loop=event_loop)
+                p._logger = logger
+                p._logger.info = MagicMock()
+                p._logger.error = MagicMock()
+                p._storage = MagicMock(spec=StorageClient)
+                p._readings_storage = MagicMock(spec=ReadingsStorageClient)
+                audit = p._audit
+                with patch.object(p._readings_storage, 'purge', side_effect=self.store_purge):
+                    with patch.object(audit, 'information', return_value=mock_audit_info()):
+                        # Test the code block when purge failed because of invalid configuration
+                        with pytest.raises(ValueError):
+                            p.purge_data(conf)
+                        p._logger.error.assert_called_with('Configuration item {} bla should be integer!'.
+                                                           format(expected_error_key))
+
+    def test_run(self, event_loop):
+        """Test that run calls all units of purge process"""
+        @asyncio.coroutine
+        def mock_audit_info():
+            return ""
```

Comment:
coroutine is unused?

Reply:
returns "" but not unused, patch uses this async coroutine